src/main/java/org/gridsuite/study/server/DataSourceConfig.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright (c) 2020, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package org.gridsuite.study.server;
+
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+
+import javax.sql.DataSource;
+
+import static org.gridsuite.study.server.StudyException.Type.DRIVER_NOT_SUPPORTED;
+
+/**
+ * @author Chamseddine Benhamed <chamseddine.benhamed at rte-france.com>
+ */
+
+@Configuration
+@PropertySource(value = {"classpath:database.properties"})
+@PropertySource(value = {"file:/config/database.properties"}, ignoreResourceNotFound = true)
+public class DataSourceConfig {
+    @Bean
+    public DataSource getDataSource(Environment env) {
+
+        String driver = env.getRequiredProperty("driver");
+        DataSourceBuilder dataSourceBuilder = DataSourceBuilder.create();
+
+        if (driver.equals("h2")) {
```

Comment:
It's weird to hardcode driver names like this, and especially when hardcoding in the production code some things for the tests. Try to avoid it

Reply:
there's a way to avoid this but with other micro-services which use R2DBC we can't (we cannot create a custom datasource bean, we should create connection factories instead!), so to share the same database.properties in deployment later i did the same logic. so either i create a specific database.properties file for micro-services which use spring data jpa and another for those which use R2DBC or i keep those changes.